### ISSUE TITLE
os-bind: bind dies and does not restart

### ISSUE THREAD

**Issue by ruffy91** [opnsense/plugins#1497]
Yesterday the BIND plugin service stopped and did not restart itself.
I can see the following in the logs just before BIND dies:
```
08-Sep-2019 14:01:29.753	general: critical: exiting (due to assertion failure)
08-Sep-2019 14:01:29.753	general: critical: #7 0x0 in ??
08-Sep-2019 14:01:29.753	general: critical: #6 0x3e007b0dc36 in ??
08-Sep-2019 14:01:29.753	general: critical: #5 0x3b13830d1ed in ??
08-Sep-2019 14:01:29.753	general: critical: #4 0x3b138244169 in ??
08-Sep-2019 14:01:29.753	general: critical: #3 0x3b13823b04c in ??
08-Sep-2019 14:01:29.753	general: critical: #2 0x3b138234728 in ??
08-Sep-2019 14:01:29.753	general: critical: #1 0x3b1382ed18a in ??
08-Sep-2019 14:01:29.753	general: critical: #0 0x3b138102120 in ??
08-Sep-2019 14:01:29.753	general: critical: resolver.c:4917: INSIST(dns_name_issubdomain(&fctx->name, &fctx->domain)) failed, back trace
08-Sep-2019 14:01:29.745	lame-servers: info: chase DS servers resolving 'd10u1qvpabtlks.cloudfront.net/DS/IN': 1.0.0.1#53
08-Sep-2019 14:01:29.514	lame-servers: info: chase DS servers resolving 'd10u1qvpabtlks.cloudfront.net/DS/IN': 1.1.1.1#53
```

This is in OPNsense 19.7.3 with os-bind 1.7

**Comment by mimugmail** [comment]
Sounds like an upstream problem, there is a pending update for bind itself if I remember correctly.

**Comment by ruffy91** [comment]
Could't find anything upstream concerning this error.
I will try to open an issue there.

In the meantime I have the following questions:

- BIND is not shown on the OPNsense Dashboard which makes it harder to diagnose the problem. How can I get it to show up?

- Can't the plugin register somewhere so that it is restarted by OPNsense/ a supervisor (sysutils/daemontools) if it dies?

**Comment by ruffy91** [comment]
Upstream issue opened:
https://gitlab.isc.org/isc-projects/bind9/issues/1219

I am unsure if this will lead to a solution however, as I am not able to reproduce the problem.

**Comment by mimugmail** [comment]
Dashboard service is called named?
Restart can be done via Monit :)

**Comment by ruffy91** [comment]
> Dashboard service is called named?
Yes thank you. Now I am unsure why I didn't see the red Stop symbol.
> Restart can be done via Monit :)
OK, i added the service and a check on udp and tcp for dns

Also the BIND people ask if I have core and debugging symbols (detached or as unstripped binary).
Are they available on onpsense?

**Comment by ruffy91** [comment]
Ok, the BIND maintainers are concerned about this issue because they already had multiple cases but am unable to find the problem even in code reviews. They are interested in a core dump, even better with debug symbols.

I can't seem to find a .core for named.
I found out that the process dies with Signal  6 SIGABRT, which according to https://backtrace.io/blog/backtrace/whats-a-coredump/ should generate a core dump.
Also ulimit shows unlimited, which afaik also says that core dumps should be generated.

How can I get OPNsense to generate the dump should named fail again?

**Comment by mimugmail** [comment]
If there are any, then in /var/core
Not sure if bind needs an option for this.

How often does this happen?

**Comment by mimugmail** [comment]
Did you delete the issue at bind gitlab? It's not there anymore ...

**Comment by ruffy91** [comment]
It was set to private because it's probably a DoS-able vulnerability and they already had 2 reports but without crash dump or debug symbols of this error when I reported it to them.
As I do not know where to find an option to create crash dumps and cannot find the debug symbols I cannot help them too.

**Comment by ruffy91** [comment]
Issue has been solved upstream in 9.14.7 and 9.15.5 and was assigned CVE-2019-6476

**Comment by mimugmail** [comment]
Close? I think it is in 19.7.6
